test/Microsoft.ML.Benchmarks/BigramAndTrigramBenchMark.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.ML.Benchmarks
+{
+    // Adding this class to not print anything to the console.
+    // This is required for the current version of BenchmarkDotNet
+    internal class EmptyWriter : TextWriter
+    {
+        internal static readonly EmptyWriter Instance = new EmptyWriter();
+        public override Encoding Encoding => null;
+    }
+
+    public class BigramAndTrigramBenchmark
+    {
+        private string _dataPath_Wiki;
+        private string _modelPath_Wiki;
+
+        [GlobalSetup(Targets = new string[] { nameof(Preceptron_CV), nameof(LightGBM_CV) })]
+        public void Setup_Preceptron_LightGBM()
+        {
+            _dataPath_Wiki = Path.GetFullPath(TestDatasets.WikiDetox.trainFilename);
+
+            if (!File.Exists(_dataPath_Wiki))
+            {
+                throw new FileNotFoundException($"Could not find {_dataPath_Wiki} Please ensure you have run 'build.cmd -- /t:DownloadExternalTestFiles /p:IncludeBenchmarkData=true' from the root");
+            }
+        }
+
+        [GlobalSetup(Target = nameof(WikiDetox))]
+        public void Setup_WikiDetox()
+        {
+            Setup_Preceptron_LightGBM();
+            _modelPath_Wiki = Path.Combine(Directory.GetCurrentDirectory(), @"WikiModel.zip");
+            string cmd = @"CV k=5 data=" + _dataPath_Wiki + " loader=TextLoader{quote=- sparse=- col=Label:R4:0 col=rev_id:TX:1 col=comment:TX:2 col=logged_in:BL:4 col=ns:TX:5 col=sample:TX:6 col=split:TX:7 col=year:R4:3 header=+} xf=Convert{col=logged_in type=R4} xf=CategoricalTransform{col=ns} xf=TextTransform{col=FeaturesText:comment wordExtractor=NGramExtractorTransform{ngram=2}} xf=Concat{col=Features:FeaturesText,logged_in,ns} tr=OVA{p=AveragedPerceptron{iter=10}} out={" + _modelPath_Wiki + "}";
+            using (var tlc = new TlcEnvironment(verbose: false, sensitivity: MessageSensitivity.None, outWriter: EmptyWriter.Instance))
+            {
+                Maml.MainCore(tlc, cmd, alwaysPrintStacktrace: false);
+            }
+        }
+
+        [Benchmark]
+        public void Preceptron_CV()
```

Comment:
How does this format sound: `[mode]_[task]_[dataset]_[featurization]_[learner]`

Mode is one of: `CV` / `TrainTest` / `Test` / etc
Task is one of: `MulticlassClassification` / `Regression` / `BinaryClassification` / `Ranking` / `AnomalyDetection` / `Clustering` / `MultiOutputRegression` / etc

So our benchmark names would then be:
* CV_Multiclass_WikiDetox_BigramsAndTrichar_OVAAveragedPerceptron
* CV_Multiclass_WikiDetox_BigramsAndTrichar_LightGBMMulticlass
* CV_Multiclass_WikiDetox_WordEmbeddings_OVAAveragedPerceptron
* CV_Multiclass_WikiDetox_WordEmbeddings_SDCAMC
* Test_Multiclass_WikiDetox_BigramsAndTrichar_OVAAveragedPerceptron _(this could better convey that we are specifically benchmarking the bulk scoring speed, whereas the above benchmarks mainly training speeds)_

We'll need a naming convention as we expand to additional datasets and pipelines.